hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java

```diff
@@ -170,7 +170,7 @@ protected HoodieCombineFileInputFormatShim createInputFormatShim() {
         if (job.get(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS, "").isEmpty()) {
           List<String> partitions = new ArrayList<>(part.getPartSpec().keySet());
           if (!partitions.isEmpty()) {
-            String partitionStr = String.join(",", partitions);
```

Comment:
hmm..very strange. How does this work in the first place? Was this separator changed between Hive2 and Hive3?

Reply:
@garyli1019 , I checked previous versions of Hive, they all use slash for split.  this pr is inspired by HUDI-314, but i think that pr does not consider mor increamtal query.    In fact, this bug only occurs when you use HoodieCombineHiveInputFormat  to query mor table.

Reply:
I see. Thanks for the explanation. Would that be possible to cover this case in the UT? Probably we could cover this case in your other PRs cause you already added a test case for this `HoodieCombineHiveInputFormat`?

Reply:
ok,  i will try. thanks  @garyli1019

Reply:
@garyli1019 ，  UT added。 could you review this pr again， thanks

Reply:
I am just getting started to understand the query side (and hence not very conversant). I tried looking in hive for CombineHiveInputFormat to find the usage of delimiter, but couldn't find this piece of code. Would you mind pointing me to file where I can find this code snippet in hive repo.

Reply:
@nsivabalan   just see the funtion initObjectInspector in MapOperator.java(my hive version is hive 3.1.1):

    // Next check if this table has partitions and if so
    // get the list of partition names as well as allocate
    // the serdes for the partition columns
**line 189**    String pcols = overlayedProps.getProperty(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS);

**line 191**    if (pcols != null && pcols.length() > 0) {
**line 192**    String[] partKeys = pcols.trim().split("/");

Reply:
👍